export.js

```diff
@@ -0,0 +1,161 @@
+function grabExcelValues(rows, artifact, objTemplate, customFieldRange) {
+    if (rows === null) {
+        getRows(artifact, objTemplate, customFieldRange);
+    }
+    else {
+        return Excel.run(function (context) {
+            let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+            let inputRange = columnRanges[artifact] + rows;
+            let customRange = (customFieldRange[0] + 3) + ":" + (customFieldRange[1] + rows);
+            let inputValues = sheet.getRange(inputRange);
+            let customFields = sheet.getRange(customRange);
+            inputValues.load();
+            customFields.load();
+            return context.sync()
+                .then(function () {
+                    customFieldObjArr = customFields.values.map(customFieldObjCreate);
+                    ajaxExport(inputValues.values, artifact, objTemplate, customFieldObjArr);
+                })
+        });
+    }
+}
+
+function getRows(artifact, objTemplate, customFieldRange) {
```

Comment:
Because of the way the office API works I don't have access to values.length without a separate load(); and handling that information in a .then() ):